enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+  state should not cause the alerts to be disabled.
+- If the user intentionally disables IO, the alerts should never fire. If IO is
+  disabled while some of the Insights alerts are firing, the alerts should
+  immediately stop firing (a part of their condition will be that IO is
+  enabled).
+- If IO was disabled by an accident, the user will be notified via the existing
+  `InsightsDisabled` alert, which is defined as a part of the Insights Operator.
+
+**Unhealthy Smart Proxy scenario:**
+
+- The latest successfully fetched and parsed report is too old
+  (`Meta.LastCheckedAt` is more than 8 hours ago)
+  - Reset the Insights recommendation metrics to avoid showing alerts about
+    recommendations that are no longer hitting.
+- Unable to fetch a report from Smart Proxy (endpoint unreachable, auth error, etc.)
+  - Log error, do not touch the Insights recommendation metrics.
+- The fetched report cannot be decoded into the expected structure, or the data
+  are otherwise invalid
+  - Log error, do not touch the Insights recommendation metrics.
+
+If there is anything wrong with the Smart Proxy or the reports returned by it,
+the metrics will not be modified until the validity of the last report expires.
+If IO is enabled, but something is wrong with the gathering or uploading, the
+metrics will not be directly affected for as long as the last report is valid.
+If IO has been disabled, or crashing for more then 10 minutes, the alerts will
+be disabled to avoid presenting the user with outdated information.
+
+### Risks and Mitigations
+
+- Disabling and re-enabling an alert (e.g, due to IO or Smart Proxy outage)
+  can trigger user-defined actions, even though there has not been any actual
+  change to the status of the recommendation. This could be very annoying if
+  it were to happen often or repeatedly over a short period of time.
+- The URL to detailed information about an active Insights recommendation will
+  be hard-coded into the alert template or the Insights Operator. If the
+  endpoint moves elsewhere, the URL will be broken and an updated version of the
+  alert template will have to be backported. The process can take several
+  months, and there is no guarantee that the user will perform the update.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Which operator conditions should fall under the unhealthy Insights Operator
+   state?
+1. How long should IO wait before considering a report to be no longer valid
+   (outdated)?
+1. If uploads keep failing, how and when will the alerts be disabled? (because
+   the current design assumes that the metrics are only modified after receiving
+   a report after a successful upload)
+
+### Test Plan
+
+Due to its nature, this feature will probably be best tested using e2e tests
+that involve other components. The Prometheus stack will be necessary to
+evaluate the metrics and alerts.
+
+The e2e tests could work as follows:
+- break some property of the cluster,
```

Comment:
Alternatively, public rule for "you look like an OCP CI cluster", and then we add something to an insights e2e-operator and/or an openshift/origin suite that says "yup, Insights is telling my I'm an OCP CI cluster"?

Reply:
That would probably work as well, but I think breaking some generally available cluster configuration might be easier and will resemble the end-user-experience more closely. I feel like this would add additional complexity to an otherwise fairly simple feature.

Reply:
Seems simpler to me than finding a reliable way to break something.  But 🤷 , it will be your CI, so if you would rather maintain a cluster-breaking workflow than a CI-focused insights rule and test-case looking for that rule, I have no problem with that.

Reply:
Well, my concern was that the creation of another rule tends to be a lengthy process. It doesn't have to be a cluster-breaking workflow, per se, as there are recommendations even for some fairly minor things. Either way, I would ask around to see which option would be easier to implement and maintain into the future, and I will keep your suggestion in mind.